Gems/ROS2/Code/Source/FactorySimulation/ConveyorBeltComponentKinematic.cpp

```diff
@@ -0,0 +1,306 @@
+                sceneInterface->RegisterSceneSimulationFinishHandler(defaultSceneHandle, m_sceneFinishSimHandler);
+                const auto& [startPoint, endPoint] = GetStartAndEndPointOfBelt(splinePtr);
+                m_startPoint = startPoint;
+                m_endPoint = endPoint;
+                m_splineLength = GetSplineLength(splinePtr);
+
+                // initial segment population
+                const float normalizedDistanceStep = 0.5f * m_segmentLength / m_splineLength;
+                for (float f = 0.f; f < 1.f; f += normalizedDistanceStep)
+                {
+                    m_ConveyorSegments.push_back(CreateSegment(splinePtr, physicsSystem, defaultSceneHandle, f));
+                }
+                AZ_Printf("ConveyorBeltComponentKinematic", "Initial Number of segments: %d", m_ConveyorSegments.size());
+
+                m_initialized = true;
+            }
+        }
+
+        // spawn first segment if needed
+        if (!m_ConveyorSegments.empty())
+        {
+            AZ::Vector3 position = GetLocationOfSegment(defaultSceneHandle, m_ConveyorSegments.front().second);
+            AZ::Vector3 distance = m_startPoint - position;
+            if (distance.GetLength() > m_segmentLength / 2.f)
+            {
+                m_ConveyorSegments.push_front(CreateSegment(m_splineConsPtr, physicsSystem, defaultSceneHandle, 0.f));
+            }
+        }
+
+        // despawn segments that are at the end of the spline
+        for (auto& [pos, handle] : m_ConveyorSegments)
+        {
+            if (pos > 1.0f)
+            {
+                sceneInterface->RemoveSimulatedBody(defaultSceneHandle, handle);
+                handle = AzPhysics::InvalidSimulatedBodyHandle;
+            }
+        }
+
+        // clear object handle from cache
+        m_ConveyorSegments.erase(
+            AZStd::remove_if(
+                m_ConveyorSegments.begin(),
+                m_ConveyorSegments.end(),
+                [](AZStd::pair<float, AzPhysics::SimulatedBodyHandle>& c)
+                {
+                    return c.second == AzPhysics::InvalidSimulatedBodyHandle;
+                }),
+            m_ConveyorSegments.end());
+
+        // move texture
+        if (m_ConveyorEntityId.IsValid())
+        {
+            AZ::Render::MaterialAssignmentId materialId;
+            AZ::Render::MaterialComponentRequestBus::EventResult(
+                materialId, m_ConveyorEntityId, &AZ::Render::MaterialComponentRequestBus::Events::FindMaterialAssignmentId, -1, "Belt");
+
+            m_textureOffset += deltaTime * m_speed * m_textureScale;
+
+            AZ::Render::MaterialComponentRequestBus::Event(
+                m_ConveyorEntityId,
+                &AZ::Render::MaterialComponentRequestBus::Events::SetPropertyValueT<float>,
+                materialId,
+                "uv.offsetU",
+                m_textureOffset);
```

Comment:
Subroutine MoveBelt could be great here, incorporating spawning, despawning and caching. It could be later replaced with some other behavior (such as moving segments to start) without affecting onTick body.